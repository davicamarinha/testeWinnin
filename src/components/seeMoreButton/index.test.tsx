import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import SeeMoreButton from "./";

describe("SeeMoreButton Tests", () => {
  it("renders a snapshot", () => {
    const { container } = render(
      <SeeMoreButton onClick={() => null} disabled={false} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should render grey if is disabled is true", () => {
    const { queryByTestId } = render(
      <SeeMoreButton onClick={() => null} disabled />
    );
    expect(queryByTestId("seeMoreBtn")).toHaveClass("bg-disab");
  });

  it("should return a click", () => {
    const handleClick = jest.fn();

    render(<SeeMoreButton onClick={handleClick} disabled={false} />);

    const button = screen.getByTestId("seeMoreBtn");
    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
