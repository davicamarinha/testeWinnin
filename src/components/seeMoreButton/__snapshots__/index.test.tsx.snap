// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeeMoreButton Tests renders a snapshot 1`] = `
<div>
  <button
    class="bg-asMedium w-full flex flex-1 h-[48px] justify-center items-center rounded-[8px] text-white font-semibold text-xl"
    data-testid="seeMoreBtn"
  >
    <span
      class="text-3xl mr-4"
    >
      +
    </span>
     Ver mais
  </button>
</div>
`;
